flight/Modules/GPS/GPS.c

```diff
@@ -261,20 +268,24 @@ static void gpsTask(void *parameters)
 			}
 
 			if (res == PARSER_COMPLETE) {
-				timeNowMs = PIOS_Thread_Systime();
-				timeOfLastUpdateMs = timeNowMs;
-				timeOfLastCommandMs = timeNowMs;
+				timeOfLastUpdateMs = PIOS_Thread_Systime();
 			}
+
+			xDelay = 0;	// For now on, don't block / wait,
+					// but consume what we can from the fifo
 		}
 
 		// Check for GPS timeout
-		timeNowMs = PIOS_Thread_Systime();
-		if ((timeNowMs - timeOfLastUpdateMs) >= GPS_TIMEOUT_MS) {
+		if ((loopTimeMs - timeOfLastUpdateMs) >= GPS_TIMEOUT_MS) {
 			// we have not received any valid GPS sentences for a while.
 			// either the GPS is not plugged in or a hardware problem or the GPS has locked up.
 			uint8_t status = GPSPOSITION_STATUS_NOGPS;
 			GPSPositionStatusSet(&status);
 			AlarmsSet(SYSTEMALARMS_ALARM_GPS, SYSTEMALARMS_ALARM_ERROR);
+			/* Don't reinitialize too often. */
+			if ((loopTimeMs - timeOfConfigAttemptMs) >= GPS_TIMEOUT_MS) {
```

Comment:
Won't this always happen at the same time? Is this second nesting doing anything?